internal/clearingway/handlers.go

```diff
@@ -466,6 +517,103 @@ func (c *Clearingway) RemoveAll(s *discordgo.Session, i *discordgo.InteractionCr
 	}
 }
 
+func (c *Clearingway) RequestColor(s *discordgo.Session, i *discordgo.InteractionCreate) {
+	g, ok := c.Guilds.Guilds[i.GuildID]
+	if !ok {
+		fmt.Printf("Interaction received from guild %s with no configuration!\n", i.GuildID)
+		return
+	}
+
+	// Ignore messages not on the correct channel
+	if i.ChannelID != g.ChannelId {
+		fmt.Printf("Ignoring message not in channel %s.\n", g.ChannelId)
+	}
+
+	err := discord.StartInteraction(s, i.Interaction, "Checking for respective clear role...")
+	if err != nil {
+		fmt.Printf("Error sending Discord message: %v\n", err)
+		return
+	}
+
+	wantedUltimate := i.ApplicationCommandData().Options[0].StringValue()
+
+	// Get all possible roles and the encounter while we're at it
+	var colorRoles []*Role
+	var wantedEncounter *Encounter
+	for _, e := range UltimateEncounters.Encounters {
+		encounter := g.Encounters.ForName(e.Name)
+		colorRoles = append(colorRoles, encounter.Roles[ColorRole])
+		if e.Name == wantedUltimate {
+			wantedEncounter = encounter
+		}
+	}
+	
+	requestedColorRole := wantedEncounter.Roles[ColorRole]
+	clearedRole := wantedEncounter.Roles[ClearedRole]
+	
+	var roleToRemove *Role
+	var cleared = false
+	for _, memberRole := range i.Member.Roles {
+		if roleToRemove == nil {
+			for _, colorRole := range colorRoles {
+				if colorRole.DiscordRole.ID == memberRole {
+					roleToRemove = colorRole
+					break
+				}
+			}
+		}
+		if (!cleared) && (memberRole == clearedRole.DiscordRole.ID) {
+			cleared = true
+		}
+	}
+
+	if cleared {
+		// remove existing color role
+		tempstr := ""
+		if roleToRemove != nil {
+			fmt.Printf("Removing role: %+v\n", roleToRemove.Name)
+			err = roleToRemove.RemoveFromCharacter(g.Id, i.Member.User.ID, c.Discord.Session)
+			if err != nil {
+				fmt.Printf("Error removing role: %+v\n", err)
+				return
+			}
+			tempstr += fmt.Sprintf("Successfully removed role: <@&%v>", roleToRemove.DiscordRole.ID)
+		}
```

Comment:
i'm kind of surprised that there's no helper function to being with. let's make this a helper